paddle/framework/op_registry.h

```diff
@@ -167,40 +114,25 @@ Add a mark to which output is temporary is helpful for future optimization.
   OpProto* proto_;
   OpAttrChecker* op_checker_;
   bool validated_{false};
-  bool has_multiple_input_{false};
-  bool has_multiple_output_{false};
-  bool has_temporary_output_{false};
 };
 
 class OpRegistry {
   using OpCreator = std::function<OperatorBase*()>;
-  using VarIndexMap = std::unordered_map<std::string, int>;
-  using VarNameList = std::vector<std::string>;
+  using VarNameMap = std::map<std::string, std::vector<std::string>>;
```

Comment:
Put this typedef to `OperatorBase` is a better idea. Because it is used by `OperatorBase::inputs_/outputs_`